pennylane/_qubit_device.py

```diff
@@ -640,7 +632,13 @@ def statistics(
                 result = self._asarray(qml.math.squeeze(samples))
 
             elif isinstance(m, CountsMP):
-                result = self.sample(obs, shot_range=shot_range, bin_size=bin_size, counts=True)
+                result = self.sample(
+                    obs,
```

Comment:
So obs could be either an observable or a measurement process?

Reply:
no, it should still be an observable to match the other functions and not break the API. The observable eventually gets passed to `_samples_to_counts`, and on master this functions [depends on obs.return_type](https://github.com/PennyLaneAI/pennylane/blob/b7d6d6d184ef0b29256c13a4071f0a513c89b20c/pennylane/_qubit_device.py#L1446). I could maybe update the function to support both to avoid introducing breaking changes, but I'm worried that it'll cause other things to break

Reply:
then again, only lightning overrides it. so I suppose we could make it support measurements or observables. either way... something has to change in lightning. I'd rather not support a Union-type because it feels sketchier (aka something only Python would let you do) but I can be convinced

Reply:
So what if the measurement is `qml.sample()` without an observable?

Reply:
oh crap good point. it already expects either one of them! updating, one sec.

Reply:
it's updated. now I pass `m` to `self.sample`. in the past, if the measurement didn't _have_ an observable, it would have received the same object anyway, just under the name `obs` (which we defined by `obs = m` because `m.obs` was None). the private `_samples_to_counts` (the original source of trouble here) has been updated to expect measurement objects no matter what now (we know it will be a measurement because it's only called `if isinstance(m, CountsMP)`